### PR TITLE
`create_test.py`: Add type hints and refactor to use pathlib

### PR DESCRIPTION
**Pull request by kenodegard**
### Description

`create_test.py` is where a package's test scripts are generated. Here we add type hints and start to replace `os.path` logic with `pathlib`. Also update docs to reflect what has been happening in the code.

### Checklist - did you ...

- [ ] ~Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?~
     
- [x] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

**Review by beeankha — DISMISSED** [review]

File: conda_build/create_test.py [hunk 1]
```diff
@@ -3,44 +3,42 @@
...
+from pathlib import Path
 
-from conda_build.utils import copy_into, ensure_list, glob, on_win, rm_rf
+from .metadata import MetaData
+from .utils import copy_into, ensure_list, on_win, rm_rf
 
 
-def create_files(m, test_dir=None):
-    """
-    Create the test files for pkg in the directory given.  The resulting
```
Comment (dholth) [inline conda_build/create_test.py:17]: Where are the test files created now?

Comment (kenodegard) [reply conda_build/create_test.py:17]: not sure what you mean? the docstring was updated and now reads:

```
Copy all test files from recipe over into testing directory.
```

Comment (dholth) [reply conda_build/create_test.py:17]: Doesn't this call some other functions that create e.g. a Python script that tries to import things the recipe asks for, and then copies them over?

**Review by beeankha — APPROVED** [review]

File: conda_build/create_test.py (hunk 1 above)
Comment (kenodegard) [reply conda_build/create_test.py:17]: No, `create_files` doesn't _create_ any new files, it only copies existing ones. `create_python_files` (and similarly named functions for other supported languages) are where the test scripts you're referencing are created.

**Pull request merged**